src/aleph_vrf/coordinator/vrf.py

```diff
@@ -292,3 +312,27 @@ async def publish_data(
         )
 
     return message.item_hash
+
+
+async def get_existing_vrf_message(
+    aleph_client: AuthenticatedAlephClient,
+    request_id: str,
+) -> Optional[PostMessage]:
+    channel = f"vrf_{request_id}"
+    ref = f"vrf_{request_id}"
+
+    logger.debug(
+        f"Getting VRF messages on {aleph_client.api_server} from request id {request_id}"
+    )
+
+    messages, status = await aleph_client.get_messages(
+        message_type=MessageType.post,
+        channels=[channel],
+        refs=[ref],
+    )
+
+    if not messages:
+        logger.debug(f"Existing VRF message for request id {request_id} not found")
+        return None
+
+    return messages[0]
```

Comment:
What happens if there are multiple messages matching the query ?

```suggestion
    if messages:
        if len(messages) > 1:
            logger.warn(f"Multiple VRF messages found for request id {request_id}")
        return messages[0]
    else:
        logger.debug(f"Existing VRF message for request id {request_id} not found")
        return None
```